`LinearVerbosity` introduces JET regression

**Describe the example**

#756 Introduced a regression in my JET tests, showing up somewhere in the `geodesic_acceleration` code used within `LevenbergMarquardt` of NonlinearSolveFirstOrder.jl.

**Minimal Reproducible Example 👇**

See [this CI](https://github.com/RomeoV/RunwayLib.jl/actions/runs/18923957350/job/54026757481?pr=106). 

**Error & Stacktrace ⚠️**

Here's a few lines of the callstack.

```
│  ┌ Warning: Solving Linear System A::LinearAlgebra.Symmetric{Float64, Matrix{Float64}} x::StaticArraysCore.SVector{6, Float64} = b::StaticArraysCore.SVector{6, Float64} is not \
│  │            properly supported. Converting `x` to a mutable array. Check the return type \
│  │            of the nonlinear function provided for optimal performance.
│  └ @ NonlinearSolveBase ~/.julia/packages/NonlinearSolveBase/aLDVk/src/linear_solve.jl:103
└  
Precompilation took 72.0874417 s
Running: setenv(`/opt/hostedtoolcache/julia/1.12.1/x64/bin/julia -C native -J/opt/hostedtoolcache/julia/1.12.1/x64/lib/julia/sys.so -g1 --startup-file=no --history-file=no --project=.. --output-o /tmp/jl_YJd3wD/image.o.a --output-incremental=no --strip-ir --strip-metadata --experimental --trim=safe /home/runner/.julia/packages/JuliaC/L13jU/src/scripts/juliac-buildscript.jl --scripts-dir /home/runner/.julia/packages/JuliaC/L13jU/src/scripts --source /home/runner/work/RunwayLib.jl/RunwayLib.jl/juliac/loadrunwaylib.jl --output-lib --compile-ccallable`,["MFLAGS=", "PATH=/opt/hostedtoolcache/julia/1.12.1/x64/bin:/snap/bin:/home/runner/.local/bin:/opt/pipx_bin:/home/runner/.cargo/bin:/home/runner/.config/composer/vendor/bin:/usr/local/.ghcup/bin:/home/runner/.dotnet/tools:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin", "DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1", "GITHUB_RUN_NUMBER=263", "GITHUB_REF_NAME=106/merge", "RUNNER_ARCH=X64", "MAKEFLAGS=", "DEPWARN=yes", "ACCEPT_EULA=Y", "ANT_HOME=/usr/share/ant", "GITHUB_REPOSITORY_ID=1037453127", "GITHUB_ACTOR=RomeoV", "GITHUB_WORKFLOW_REF=RomeoV/RunwayLib.jl/.github/workflows/CI.yml@refs/pull/106/merge", "ALLOW_RERESOLVE=true", "ANDROID_NDK_LATEST_HOME=/usr/local/lib/android/sdk/ndk/28.2.13676358", "SYSTEMD_EXEC_PID=1876", "OPENBLAS_DEFAULT_NUM_THREADS=1", "RUNNER_ENVIRONMENT=github-hosted", "USER=runner", "CONDA=/usr/share/miniconda", "FORCE_LATEST_COMPATIBLE_VERSION=auto", "JULIA_PKG_SERVER_REGISTRY_PREFERENCE=eager", "GOROOT_1_23_X64=/opt/hostedtoolcache/go/1.23.12/x64", "GITHUB_REF_PROTECTED=false", "GITHUB_SHA=07010840bd3593c2b7fead31277bcae81090cc31", "JAVA_HOME=/usr/lib/jvm/temurin-17-jdk-amd64", "GITHUB_API_URL=https://api.github.com/", "GITHUB_RUN_ATTEMPT=1", "GITHUB_ACTIONS=true", "VCPKG_INSTALLATION_ROOT=/usr/local/share/vcpkg", "MAKELEVEL=1", "ANDROID_SDK_ROOT=/usr/local/lib/android/sdk", "SWIFT_PATH=/usr/share/swift/usr/bin", "GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_95d7c526-e2d3-458d-8e0a-a0c836168a88", "SHELL=/bin/bash", "JAVA_HOME_17_X64=/usr/lib/jvm/temurin-17-jdk-amd64", "GITHUB_ACTION_PATH=/home/runner/work/_actions/julia-actions/julia-runtest/v1", "RUNNER_NAME=GitHub Actions 1000003407", "EDGEWEBDRIVER=/usr/local/share/edge_driver", "GHCUP_INSTALL_BASE_PREFIX=/usr/local", "GITHUB_RUN_ID=18923957350", "ANNOTATE=false", "NVM_DIR=/home/runner/.nvm", "ImageOS=ubuntu24", "GITHUB_HEAD_REF=fix-condapkg-ci-issues", "GITHUB_RETENTION_DAYS=90", "GITHUB_SERVER_URL=https://github.com/", "GITHUB_OUTPUT=/home/runner/work/_temp/_runner_file_commands/set_output_95d7c526-e2d3-458d-8e0a-a0c836168a88", "GITHUB_JOB=test", "DEBIAN_FRONTEND=noninteractive", "RUNNER_TRACKING_ID=github_f65b806b-5ca2-4868-b958-1d9e0048b690", "RUNNER_TOOL_CACHE=/opt/hostedtoolcache", "HOMEBREW_CLEANUP_PERIODIC_FULL_DAYS=3650", "AZURE_EXTENSION_DIR=/opt/az/azcliextensions", "GITHUB_ACTION_REPOSITORY=", "CHROMEWEBDRIVER=/usr/local/share/chromedriver-linux64", "GITHUB_WORKFLOW=CI", "HOMEBREW_NO_AUTO_UPDATE=1", "GITHUB_ACTION=__julia-actions_julia-runtest", "HOME=/home/runner", "JAVA_HOME_8_X64=/usr/lib/jvm/temurin-8-jdk-amd64", "GROUP=C", "GITHUB_EVENT_PATH=/home/runner/work/_temp/_github_workflow/event.json", "GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_95d7c526-e2d3-458d-8e0a-a0c836168a88", "SGX_AESM_ADDR=1", "GITHUB_REF=refs/pull/106/merge", "JAVA_HOME_25_X64=/usr/lib/jvm/temurin-25-jdk-amd64", "GITHUB_REPOSITORY=RomeoV/RunwayLib.jl", "INVOCATION_ID=3b26efd9bcaa493fabf7442848ed275f", "ImageVersion=20250929.60.1", "JAVA_HOME_21_X64=/usr/lib/jvm/temurin-21-jdk-amd64", "ANDROID_NDK=/usr/local/lib/android/sdk/ndk/27.3.13750724", "LANG=C.UTF-8", "SHLVL=1", "LOGNAME=runner", "MEMORY_PRESSURE_WATCH=/sys/fs/cgroup/system.slice/hosted-compute-agent.service/memory.pressure", "DOTNET_MULTILEVEL_LOOKUP=0", "RUNNER_WORKSPACE=/home/runner/work/RunwayLib.jl", "XDG_RUNTIME_DIR=/run/user/1001", "GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_95d7c526-e2d3-458d-8e0a-a0c836168a88", "GITHUB_GRAPHQL_URL=https://api.github.com/graphql", "GOROOT_1_22_X64=/opt/hostedtoolcache/go/1.22.12/x64", "GITHUB_BASE_REF=master", "_=/opt/hostedtoolcache/julia/1.12.1/x64/bin/julia", "GRADLE_HOME=/usr/share/gradle-9.1.0", "GITHUB_ACTION_REF=", "GOROOT_1_24_X64=/opt/hostedtoolcache/go/1.24.7/x64", "PIPX_HOME=/opt/pipx", "GITHUB_REPOSITORY_OWNER_ID=8644490", "GITHUB_ACTOR_ID=8644490", "ANDROID_NDK_ROOT=/usr/local/lib/android/sdk/ndk/27.3.13750724", "MEMORY_PRESSURE_WRITE=c29tZSAyMDAwMDAgMjAwMDAwMAA=", "***", "ACTIONS_RUNNER_ACTION_ARCHIVE_CACHE=/opt/actionarchivecache", "ENABLE_RUNNER_TRACING=true", "GITHUB_WORKSPACE=/home/runner/work/RunwayLib.jl/RunwayLib.jl", "COMPILED_MODULES=yes", "GITHUB_WORKFLOW_SHA=07010840bd3593c2b7fead31277bcae81090cc31", "CHROME_BIN=/usr/bin/google-chrome", "ANDROID_HOME=/usr/local/lib/android/sdk", "CI=true", "XDG_CONFIG_HOME=/home/runner/.config", "COVERAGE=true", "POWERSHELL_DISTRIBUTION_CHANNEL=GitHub-Actions-ubuntu24", "GECKOWEBDRIVER=/usr/local/share/gecko_driver", "GITHUB_PATH=/home/runner/work/_temp/_runner_file_commands/add_path_95d7c526-e2d3-458d-8e0a-a0c836168a88", "RUNNER_OS=Linux", "JOURNAL_STREAM=9:14711", "CHECK_BOUNDS=yes", "GITHUB_REF_TYPE=branch", "BOOTSTRAP_HASKELL_NONINTERACTIVE=1", "PIPX_BIN_DIR=/opt/pipx_bin", "RUNNER_TEMP=/home/runner/work/_temp", "SELENIUM_JAR_PATH=/usr/share/java/selenium-server.jar", "GITHUB_TRIGGERING_ACTOR=RomeoV", "JAVA_HOME_11_X64=/usr/lib/jvm/temurin-11-jdk-amd64", "GITHUB_REPOSITORY_OWNER=RomeoV", "GITHUB_EVENT_NAME=pull_request", "DOTNET_NOLOGO=1", "OPENBLAS_MAIN_FREE=1", "ANDROID_NDK_HOME=/usr/local/lib/android/sdk/ndk/27.3.13750724", "AGENT_TOOLSDIRECTORY=/opt/hostedtoolcache"])
Compiling...
Verifier error #1: unresolved call from statement SciMLLogging.logging_message_level(Base.getfield(Base.getfield(cache::Any, :verbose)::LinearSolve.LinearVerbosity, :solver_failure)::Any)::Any
Stacktrace:
  [1] macro expansion
    @ ~/.julia/packages/SciMLLogging/bdVXg/src/utils.jl:123 [inlined]
  [2] macro expansion
    @ ~/.julia/packages/LinearSolve/ZgNCX/src/factorization.jl:10 [inlined]
  [3] solve!(cache::LinearSolve.LinearCache{LinearAlgebra.Symmetric{Float64, Matrix{Float64}}, Vector{Float64}, StaticArraysCore.MVector{3, Float64}, SciMLBase.NullParameters, LinearSolve.CholeskyFactorization{LinearAlgebra.NoPivot, Nothing}, LinearAlgebra.Cholesky{Float64, Matrix{Float64}}, SciMLOperators.IdentityOperator, SciMLOperators.IdentityOperator, Float64, Bool, LinearSolve.LinearSolveAdjoint{Missing}}, alg::LinearSolve.CholeskyFactorization{LinearAlgebra.NoPivot, Nothing}; kwargs::@Kwargs{})
    @ LinearSolve ~/.julia/packages/LinearSolve/ZgNCX/src/factorization.jl:1
  [4] solve!(::LinearSolve.LinearCache{LinearAlgebra.Symmetric{Float64, Matrix{Float64}}, Vector{Float64}, StaticArraysCore.MVector{3, Float64}, SciMLBase.NullParameters, LinearSolve.CholeskyFactorization{LinearAlgebra.NoPivot, Nothing}, LinearAlgebra.Cholesky{Float64, Matrix{Float64}}, SciMLOperators.IdentityOperator, SciMLOperators.IdentityOperator, Float64, Bool, LinearSolve.LinearSolveAdjoint{Missing}}; kwargs::@Kwargs{})
    @ LinearSolve ~/.julia/packages/LinearSolve/ZgNCX/src/common.jl:441 [inlined]
  [5] solve!(::LinearSolve.LinearCache{LinearAlgebra.Symmetric{Float64, Matrix{Float64}}, Vector{Float64}, StaticArraysCore.MVector{3, Float64}, SciMLBase.NullParameters, LinearSolve.CholeskyFactorization{LinearAlgebra.NoPivot, Nothing}, LinearAlgebra.Cholesky{Float64, Matrix{Float64}}, SciMLOperators.IdentityOperator, SciMLOperators.IdentityOperator, Float64, Bool, LinearSolve.LinearSolveAdjoint{Missing}})
    @ LinearSolve ~/.julia/packages/LinearSolve/ZgNCX/src/common.jl:440 [inlined]
  [6] (::NonlinearSolveBase.LinearSolveJLCache{LinearSolve.LinearCache{LinearAlgebra.Symmetric{Float64, Matrix{Float64}}, Vector{Float64}, StaticArraysCore.MVector{3, Float64}, SciMLBase.NullParameters, LinearSolve.CholeskyFactorization{LinearAlgebra.NoPivot, Nothing}, LinearAlgebra.Cholesky{Float64, Matrix{Float64}}, SciMLOperators.IdentityOperator, SciMLOperators.IdentityOperator, Float64, Bool, LinearSolve.LinearSolveAdjoint{Missing}}, LinearSolve.CholeskyFactorization{LinearAlgebra.NoPivot, Nothing}})(; A::LinearAlgebra.Symmetric{Float64, Matrix{Float64}}, b::Vector{Float64}, linu::StaticArraysCore.SVector{3, Float64}, reuse_A_if_factorization::Bool, verbose::Bool, kwargs::@Kwargs{})
    @ NonlinearSolveBaseLinearSolveExt ~/.julia/packages/NonlinearSolveBase/aLDVk/ext/NonlinearSolveBaseLinearSolveExt.jl:23 [inlined]
  [7] solve!(cache::NonlinearSolveBase.DampedNewtonDescentCache{Matrix{Float64}, StaticArraysCore.SVector{3, Float64}, Vector{StaticArraysCore.SVector{3, Float64}}, NonlinearSolveBase.LinearSolveJLCache{LinearSolve.LinearCache{LinearAlgebra.Symmetric{Float64, Matrix{Float64}}, Vector{Float64}, StaticArraysCore.MVector{3, Float64}, SciMLBase.NullParameters, LinearSolve.CholeskyFactorization{LinearAlgebra.NoPivot, Nothing}, LinearAlgebra.Cholesky{Float64, Matrix{Float64}}, SciMLOperators.IdentityOperator, SciMLOperators.IdentityOperator, Float64, Bool, LinearSolve.LinearSolveAdjoint{Missing}}, LinearSolve.CholeskyFactorization{LinearAlgebra.NoPivot, Nothing}}, Matrix{Float64}, Vector{Float64}, Nothing, NonlinearSolveFirstOrder.LevenbergMarquardtDampingCache{Float64, Float64, Float64, Float64, Float64, LinearAlgebra.Diagonal{Float64, StaticArraysCore.SVector{3, Float64}}, Nothing, StaticArraysCore.SMatrix{3, 3, Float64, 9}, NonlinearSolveFirstOrder.LevenbergMarquardtDampingFunction{Float64, Float64, Float64}, Float64}, Nothing, Val{false}, Val{:normal_form}}, J::Matrix{Float64}, fu::Vector{Float64}, u::StaticArraysCore.SVector{3, Float64}, idx::Val{2}; skip_solve::Bool, new_jacobian::Bool, kwargs::@Kwargs{reuse_A_if_factorization::Bool})
    @ NonlinearSolveBase ~/.julia/packages/NonlinearSolveBase/aLDVk/src/descent/damped_newton.jl:223
  [8] solve!(cache::NonlinearSolveBase.GeodesicAccelerationCache{StaticArraysCore.SVector{3, Float64}, Nothing, NonlinearSolveBase.DampedNewtonDescentCache{Matrix{Float64}, StaticArraysCore.SVector{3, Float64}, Vector{StaticArraysCore.SVector{3, Float64}}, NonlinearSolveBase.LinearSolveJLCache{LinearSolve.LinearCache{LinearAlgebra.Symmetric{Float64, Matrix{Float64}}, Vector{Float64}, StaticArraysCore.MVector{3, Float64}, SciMLBase.NullParameters, LinearSolve.CholeskyFactorization{LinearAlgebra.NoPivot, Nothing}, LinearAlgebra.Cholesky{Float64, Matrix{Float64}}, SciMLOperators.IdentityOperator, SciMLOperators.IdentityOperator, Float64, Bool, LinearSolve.LinearSolveAdjoint{Missing}}, LinearSolve.CholeskyFactorization{LinearAlgebra.NoPivot, Nothing}}, Matrix{Float64}, Vector{Float64}, Nothing, NonlinearSolveFirstOrder.LevenbergMarquardtDampingCache{Float64, Float64, Float64, Float64, Float64, LinearAlgebra.Diagonal{Float64, StaticArraysCore.SVector{3, Float64}}, Nothing, StaticArraysCore.SMatrix{3, 3, Float64, 9}, NonlinearSolveFirstOrder.LevenbergMarquardtDampingFunction{Float64, Float64, Float64}, Float64}, Nothing, Val{false}, Val{:normal_form}}, SciMLBase.NonlinearFunction{false, SciMLBase.FullSpecialize, typeof(RunwayLib.pose_optimization_objective), LinearAlgebra.UniformScaling{Bool}, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, typeof(SciMLBase.DEFAULT_OBSERVED_NO_TIME), Nothing, Nothing, Nothing, Nothing}, PoseOptimizationParams3DOF{Rotations.RotZYX{Float64}, PointFeatures{Unitful.Quantity{Float64, 𝐋, Unitful.FreeUnits{(m,), 𝐋, nothing}}, Unitful.Quantity{Float64, NoDims, Unitful.FreeUnits{(pixel,), NoDims, nothing}}, Float64, Vector{WorldPoint{Unitful.Quantity{Float64, 𝐋, Unitful.FreeUnits{(m,), 𝐋, nothing}}}}, Vector{ProjectionPoint{Unitful.Quantity{Float64, NoDims, Unitful.FreeUnits{(pixel,), NoDims, nothing}}, :offset}}, CameraMatrix{:offset, Unitful.Quantity{Float64, NoDims, Unitful.FreeUnits{(pixel,), NoDims, nothing}}}, Matrix{Float64}, LinearAlgebra.LowerTriangular{Float64, Matrix{Float64}}}, LineFeatures{Unitful.Quantity{Float64, 𝐋, Unitful.FreeUnits{(m,), 𝐋, nothing}}, Unitful.Quantity{Float64, NoDims, Unitful.FreeUnits{(pixel,), NoDims, nothing}}, Unitful.Quantity{Float64, NoDims, Unitful.FreeUnits{(rad,), NoDims, nothing}}, StaticArraysCore.SVector{0, Tuple{WorldPoint{Unitful.Quantity{Float64, 𝐋, Unitful.FreeUnits{(m,), 𝐋, nothing}}}, WorldPoint{Unitful.Quantity{Float64, 𝐋, Unitful.FreeUnits{(m,), 𝐋, nothing}}}}}, StaticArraysCore.SVector{0, Line{Unitful.Quantity{Float64, NoDims, Unitful.FreeUnits{(pixel,), NoDims, nothing}}, Unitful.Quantity{Float64, NoDims, Unitful.FreeUnits{(rad,), NoDims, nothing}}}}, CameraConfig{:offset}, StaticArraysCore.SMatrix{0, 0, Float64, 0}, StaticArraysCore.SMatrix{0, 0, Float64, 0}}}, Float64, typeof(NonlinearSolveBase.L2_NORM), Float64, Vector{Float64}, Vector{Float64}, StaticArraysCore.SVector{3, Float64}}, J::Matrix{Float64}, fu::Vector{Float64}, u::StaticArraysCore.SVector{3, Float64}, idx::Val{1}; skip_solve::Bool, kwargs::@Kwargs{new_jacobian::Bool})
    @ NonlinearSolveBase ~/.julia/packages/NonlinearSolveBase/aLDVk/src/descent/geodesic_acceleration.jl:119
```


**Additional context**

It's possible this is happening because we currently get a warning due to the mix of StaticArrays and Dense matrices (printed above). If this is confirmed to be the root cause I can close this issue again.

--- a/src/common.jl
+++ b/src/common.jl
@@ -105,7 +105,7 @@ The cache automatically tracks when matrix `A` or parameters `p` change by setti
 appropriate freshness flags. When `solve!` is called, stale cache entries are automatically
 recomputed as needed.
 """
-mutable struct LinearCache{TA, Tb, Tu, Tp, Talg, Tc, Tl, Tr, Ttol, Tlv, issq, S}
+mutable struct LinearCache{TA, Tb, Tu, Tp, Talg, Tc, Tl, Tr, Ttol, Tlv <: LinearVerbosity, issq, S}
     A::TA
     b::Tb
     u::Tu

